src/WorkerChannel.ts

```diff
@@ -2,22 +2,32 @@
 // Licensed under the MIT License.
 
 import { Context } from '@azure/functions';
+import { readJson } from 'fs-extra';
 import { AzureFunctionsRpcMessages as rpc } from '../azure-functions-language-worker-protobuf/src/rpc';
 import { IFunctionLoader } from './FunctionLoader';
 import { IEventStream } from './GrpcClient';
+import path = require('path');
 
 type InvocationRequestBefore = (context: Context, userFn: Function) => Function;
 type InvocationRequestAfter = (context: Context) => void;
 
+export interface PackageJson {
+    type?: 'commonjs' | 'module';

```

Comment:
```suggestion
    type?: string;
```

This interface needs to describe what a user _can_ do, not what they _should_ do. For example, does Node.js respect "MoDuLe"? If so, we probably should, too (aka `packageJson.type?.toLowerCase() === 'module'`)

Reply:
I don't think anything other than lowercase is _legal_ (see https://nodejs.org/docs/latest-v16.x/api/packages.html#type) but since it's JSON it's not exactly enforced.

Reply:
I changed the interface to `string`. I kept the existing logic as-is though, not respecting anything other than `"module"` all lowercase.